### PR TITLE
Implemented auth tests

### PR DESCRIPTION
**Pull request by rozza**
JAVA-2680

### CODE REVIEW

File: driver-core/src/test/unit/com/mongodb/ConnectionStringSpecification.groovy [hunk 1]
```diff
@@ -279,7 +279,7 @@ class ConnectionStringSpecification extends Specification {
         new ConnectionString('mongodb://jeff@localhost/?' +
                            'authMechanism=GSSAPI')            | createGSSAPICredential('jeff')
         new ConnectionString('mongodb://jeff:123@localhost/?' +
-                           'authMechanism=PLAIN')             | createPlainCredential('jeff', 'admin', '123'.toCharArray())
+                           'authMechanism=PLAIN')             | createPlainCredential('jeff', '$external', '123'.toCharArray())
```
Comment (rozza) [inline driver-core/src/test/unit/com/mongodb/ConnectionStringSpecification.groovy]: This is possibly a breaking change for users that used PLAIN without declaring a database or an authSource.

Comment (jyemin) [reply driver-core/src/test/unit/com/mongodb/ConnectionStringSpecification.groovy]: Worth adding to the release notes to the Upgrading section.

Comment (rozza) [reply driver-core/src/test/unit/com/mongodb/ConnectionStringSpecification.groovy]: Done!

**Comment by rozza** [comment]
I added a comment to the upgrade section and pushed.

**Pull request closed without merging**